mesonbuild/interpreter.py

```diff
@@ -1214,6 +1213,87 @@ def get_cross_property_method(self, args, kwargs):
+                       'vala_args',
+                       'd_args',
+                   ])
+
+vala_kwargs = set(['vala_gir', 'vala_header'])
+rust_kwargs = set(['rust_crate_type',
+               ])
+cs_kwargs = set(['resources'])
+
+buildtarget_kwargs = set(['sources',
+                          'link_with',
+                          'link_whole',
+                          'link_args',
+                          'link_depends',
+                          'include_directories',
+                          'dependencies',
+                          'gui_app',
+                          'extra_files',
+                          'install',
+                          'install_rpath',
+                          'install_dir',
+                          'objects',
+                          'native',
+                          'name_suffix',
+                          'name_prefix',
+                          'build_by_default',
+                          'vs_module_defs',
+                          'pic',
+                          'override_options',
+                      ])
+
+exe_kwargs = set()
+exe_kwargs.update(buildtarget_kwargs)
+exe_kwargs.update(lang_arg_kwargs)
+exe_kwargs.update(pch_kwargs)
+exe_kwargs.update(vala_kwargs)
+exe_kwargs.update(rust_kwargs)
+exe_kwargs.update(cs_kwargs)
+
+shlib_kwargs = set()
+shlib_kwargs.update(buildtarget_kwargs)
+shlib_kwargs.update(lang_arg_kwargs)
+shlib_kwargs.update(pch_kwargs)
+shlib_kwargs.update(['version', 'soversion'])
+shmod_kwargs = shlib_kwargs
+stlib_kwargs = shlib_kwargs
+
+jar_kwargs = exe_kwargs.copy()
+jar_kwargs.update(['main_class'])
+
+build_target_kwargs = exe_kwargs.copy()
+build_target_kwargs.update(['target_type'])
+
+permitted_kwargs = {'project': set(['version', 'meson_version', 'default_options', 'license', 'subproject_dir']),
+                    'executable': exe_kwargs,
+                    'shared_library': shlib_kwargs,
+                    'static_library': stlib_kwargs,
+                    'shared_module': shmod_kwargs,
+                    'jar': jar_kwargs,
+                    'build_target': build_target_kwargs,
+                    'custom_target': set(['input', 'output', 'command', 'install', 'install_dir', 'build_always', 'capture', 'depends', 'depend_files', 'depfile', 'build_by_default']),
+                    'run_target': set(['command', 'depends']),
+                    'generator': set(['arguments', 'output', 'depfile']),
+                    'test': set(['args', 'env', 'is_parallel', 'should_fail', 'timeout', 'workdir', 'suite']),
+                    'benchmark': set(['args', 'env', 'should_fail', 'timeout', 'workdir', 'suite']),
+                    'install_headers': set(['install_dir', 'subdir']),
+                    'install_man': set(['install_dir']),
+                    'install_data': set(['install_dir', 'install_mode', 'sources']),
+                    'install_subdir': set(['install_dir', 'install_mode']),
+}
```

Comment:
__[flake8]__
* [E124] closing bracket does not match visual indentation

<!-- Comment by [SideCI](https://sideci.com) -->